### PR TITLE
test: Added gcxs to benchmarks to be tested with CodSpeed

### PR DESCRIPTION
**Pull request by DeaMariaLeon**
## What type of PR is this? (check all applicable)

- [ ] 💾 Refactor
- [ ] 🪄 Feature
- [ ] 🐞 Bug Fix
- [ ] 🔧 Optimization
- [ ] 📚 Documentation
- [x] 🧪 Test
- [ ] 🛠️ Other

## Related issues

- Related issue #
- Closes #

## Checklist

- [x] Code follows style guide
- [x] Tests added
- [ ] Documented the changes

***

## Please explain your changes below.
I integrated the tests from `benchmark_gcxs.py` to `test_benchmark_coo.py`. Maybe the file name needs to be changed.

### CODE REVIEW

File: benchmarks/test_benchmark_coo.py [hunk 1]
```diff
@@ -123,3 +129,46 @@ def test_index_fancy(benchmark, indexing_args, seed):
...
+    x, t = densemul_args
+
+    u = t.T
+
+    # Numba compilation
+    u @ x
+
+    @benchmark
+    def bench():
+        u @ x
```
[the remark this answers is not in the document]
Comment (hameerabbasi) [reply benchmarks/test_benchmark_coo.py:174]: If shapes are incompatible, we can just `pytest.skip`.

**Review by hameerabbasi — COMMENTED** [review]
Thanks for the refactors here, @DeaMariaLeon! I feel the matmul benchmark can be improved, everything else looks great.

File: benchmarks/test_benchmark_coo.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply benchmarks/test_benchmark_coo.py:174]: Should `def test_matmul` be merged with these? (lines 17 - 30) @hameerabbasi ?

[the remark this answers is not in the document]
Comment (hameerabbasi) [reply benchmarks/test_benchmark_coo.py:174]: Yep!

[the remark this answers is not in the document]
Comment (DeaMariaLeon) [reply benchmarks/test_benchmark_coo.py:174]: Just for posterity: both benchmarks (on lines 153 and 164) are not the same. There is the transpose, and I was recreating the original benchmark that was already in `main`. 
On the original file `benchmark_gcxs.py`, they were creating 3 matrices. For the 3rd matrix they were inverting the shape. 
Screenshot 2024-08-20 at 09 02 31

File: benchmarks/test_benchmark_coo.py [hunk 2]
```diff
@@ -138,13 +138,18 @@ def get_densemul_id(param):
...
+
+
+@pytest.fixture(params=itertools.product([200, 500, 1000], [200, 500, 1000], [200, 500, 1000]), ids=get_sides_ids)
 def sides(request):
     m, n, p = request.param
     return m, n, p
 
 
-@pytest.fixture(params=itertools.product([0, 1], ["coo", "gcxs"]), ids=get_densemul_id)
+@pytest.fixture(params=([(0, "coo"), (0, "gcxs"), (1, "gcxs")]), ids=get_densemul_id)
```
Comment (hameerabbasi) [inline benchmarks/test_benchmark_coo.py:152]: I would pass a list here for the ids since COO doesn't really have a compressed axis.

**Review by hameerabbasi — APPROVED** [review]

**Pull request merged**